### PR TITLE
Web: Add outline-info Alert.tsx + move navigation code for general use

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport.e/issues/2998

- Add an alert type that's outlined (second time needing it so moved it to general use)
- Adds nesting to shared Navigation component (basically just moving code from Discover for general use, refactor + test)

### CODE REVIEW

**Review by ryanclark — APPROVED** [review]

File: web/packages/teleport/src/components/Wizard/Navigation/Navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+ *
+ * For nested views, it is required to apply
+ * function `addIndexToViews(views: BaseView<T>[])`
+ * before passing the views to Navigation so it can correctly
+ * increment the steps.
+ *
+ * For simple views, defining indexes is not required.
+ *
+ *
+ * Example of how nesting views are used for Discover wizards:
```
Comment (gzdunek) [inline web/packages/teleport/src/components/Wizard/Navigation/Navigation.tsx:54]: You can use `@example` tag https://jsdoc.app/tags-example

File: web/packages/teleport/src/components/Wizard/Navigation/StepItem.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    return (
+      <StepList
+        views={props.view.views}
+        currentStep={props.currentStep}
+        index={props.index}
+      />
+    );
+  }
+
+  const index = props.view.index != null ? props.view.index : props.index;
```
Comment (gzdunek) [inline web/packages/teleport/src/components/Wizard/Navigation/StepItem.tsx:53]:
```suggestion
  const index = props.view.index ?? props.index;
```

File: web/packages/teleport/src/components/Wizard/flow.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+/**
+ * findViewAtIndex will recursively loop views and their children in order to find the deepest
+ * match at that index.
+ */
+export function findViewAtIndex<T>(
+  views: BaseView<T>[],
+  currentStep: number
+): BaseView<T> | null {
```
Comment (gzdunek) [inline web/packages/teleport/src/components/Wizard/flow.tsx:83]: Nit: I don't think this function returns `null`, it rather should be `undefined`.

File: web/packages/teleport/src/Discover/Discover.tsx [hunk 4]
```diff
@@ -63,11 +63,16 @@ function DiscoverContent() {
     <>
       <FeatureBox>
         {hasSelectedResource && (
-          <Navigation
-            currentStep={currentStep}
-            views={indexedViews}
-            selectedResource={agentProps.resourceSpec}
-          />
+          <Box mt="10px" mb="45px">
```
Comment (gzdunek) [inline web/packages/teleport/src/Discover/Discover.tsx:66]: Do you think we could use theme values?

**Review by gzdunek — APPROVED** [review]

File: web/packages/design/src/Alert/Alert.jsx [hunk 5]
```diff
@@ -45,6 +46,14 @@ const kind = props => {
         background: theme.colors.success,
         color: theme.colors.text.primaryInverse,
       };
+    case 'outline-info':
+      return {
+        background: fade(theme.colors.link, 0.1),
+        border: `${theme.radii[1]}px solid ${theme.colors.link}`,
+        borderRadius: `${theme.radii[3]}px`,
```
Comment (gzdunek) [inline web/packages/design/src/Alert/Alert.jsx:53]: Having a different border radius here is on purpose?

Comment (kimlisa) [reply web/packages/design/src/Alert/Alert.jsx:53]: yes, the latest design called for it: https://www.figma.com/file/uLevdNsEnIvvLDSZ9sQqXM/Discover-Access?type=design&node-id=1963-9188&mode=design&t=9oBHExTE75bk4kJA-0

**Pull request merged**